### PR TITLE
Tweak max rating eligibility based on primary diagnostic code

### PR DESCRIPTION
**Pull request by nanotone**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- This PR excludes certain diagnostic codes from displaying a 526 Max Rating notice, based on research specific to digestive system and infectious disease diagnostic codes.
- I'm on the Employee Experience team, which works with the 526 DBEX team that maintains this component.

## Related issue(s)

- department-of-veterans-affairs/abd-vro#3230

## Testing done

- [x] *New code is covered by unit tests*
- Old behavior: all diagnostic codes were sent to the Max Rating API to determine if the Max Rating notice should be displayed.
- New behavior: infectious disease codes and certain digestive system codes are excluded.
- The DC-specific logic has been broken out into a method that now has its own RSpec tests to cover each scenario.

## What areas of the site does it impact?
* User education displayed on 526 Rated Disabilities screen; no user functionality is affected.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

File: app/services/claim_fast_tracking/max_rating_annotator.rb [hunk 1]
```diff
@@ -13,9 +14,8 @@ def self.annotate_disabilities(rated_disabilities_response)
 
       diagnostic_codes = rated_disabilities_response.rated_disabilities
                                                     .compact # filter out nil entries in rated_disabilities
+                                                    .select { |rd| eligible_for_request?(rd) } # select only eligible
```
Comment (nanotone) [inline app/services/claim_fast_tracking/max_rating_annotator.rb:17]: The `eligible_for_request?` method is more fleshed out, and now takes the entire RatedDisability object, not just the diagnostic code.

File: app/services/claim_fast_tracking/max_rating_annotator.rb [hunk 2]
```diff
@@ -64,10 +64,16 @@ def self.get_ratings(diagnostic_codes)
...
+
+      return false if EXCLUDED_DIGESTIVE_CODES.include?(rated_disability.diagnostic_code)
+
+      return true if Flipper.enabled?(:disability_526_maximum_rating_api_all_conditions)
+
+      SELECT_DISABILITIES.include?(rated_disability.diagnostic_code)
     end
 
-    private_class_method :get_ratings, :eligible_for_request?
+    private_class_method :get_ratings
```
Comment (nanotone) [inline app/services/claim_fast_tracking/max_rating_annotator.rb:77]: `eligible_for_request?` (a no-side-effects utility method) is now publicly accessible, so we can write better / more comprehensive tests for it.

File: app/services/claim_fast_tracking/max_rating_annotator.rb [hunk 3]
```diff
@@ -64,10 +64,15 @@ def self.get_ratings(diagnostic_codes)
       nil
     end
 
-    def self.eligible_for_request?(dc)
+    def self.eligible_for_request?(rated_disability)
+      return false if %i[infectious_disease missing_diagnostic_code].include?(diagnostic_code_type(rated_disability))
```
Comment (dfitchett) [inline app/services/claim_fast_tracking/max_rating_annotator.rb:68]: Mind-blowing use of symbols in ruby 🤯

**Review by dfitchett — APPROVED** [review]
LGTM

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**